src/pages/Home/index.js

```diff
@@ -2,25 +2,40 @@ import React, { useState, useEffect, useContext } from "react";
 import { SearchEngine } from "../../components/SearchEngine";
 import { Filter } from "../../components/Filter";
 import { FilmCard } from "../../components/FilmCard";
+import { UserSessionValidation } from "../../components/UserSessionValidation";
 import axios from "axios";
 import { Menu } from "../../components/Menu";
+import { Link, Navigate } from "react-router-dom";
 import "./Home.scss";
+// Context
 import { AppContext } from "../../context/AppContext";
-import { Link } from "react-router-dom";
 
 function Home() {
   // fetch Data
   const [films, setFilms] = useState([]);
-  useEffect(async () => {
-    const response = await axios.get("https://ghibliapi.herokuapp.com/films");
-    setFilms(response.data);
+
+  useEffect(() => {
+    let isSubscribed = true;
+    axios
+      .get("https://ghibliapi.herokuapp.com/films")
+      .then(response => {
+        if (isSubscribed) {
```

Comment:
Is this validation to make sure that only is executed once?

Reply:
Of course. Besides we're already receiving errors and handle them, this keep being async and is also executed once the component is rendered due the useEffect. 

That's the difference from the other.